onadata/apps/api/viewsets/openid_connect_viewset.py

```diff
@@ -129,7 +129,10 @@ def callback(self, request, **kwargs):  # pylint: disable=no-self-use
                 last_name = claim_values.get(LAST_NAME)
                 first_name = claim_values.get(FIRST_NAME)
                 if not first_name:
-                    first_name = last_name
+                    if last_name:
+                        first_name = last_name
+                    else:
+                        return HttpResponseBadRequest()
```

Comment:
Let's have a descriptive error message. The pattern established in this file should be something we should change direction by returning descriptive messages.

Probably also make it a bit terse:
```python
if not first_name and not last_name:
    raise HttpResponseBadRequest(json.dumps({"error": _("Missing required fields: %s, %s." % (FIRST_NAME, LAST_NAME)}))
else:
    first_name = first_name or last_name
```